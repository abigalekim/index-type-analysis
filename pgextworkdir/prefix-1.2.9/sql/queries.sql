-- operators
select a, b,
   a <= b as "<=", a < b as "<", a = b as "=", a <> b as "<>", a >= b as ">=", a > b as ">",
   a @> b as "@>", a <@ b as "<@", a && b as "&&"
from  (select a::prefix_range, b::prefix_range
         from (values('123', '123'),
                     ('123', '124'),
                     ('123', '123[4-5]'),
                     ('123[4-5]', '123[2-7]'),
                     ('123', '[2-3]')) as t(a, b)
      ) as x;

-- transitivity
select a, b, c, a <= b as "a <= b", b <= c as "b <= c", a <= c as "a <= c"
from  (select a::prefix_range, b::prefix_range, c::prefix_range
         from (values('123', '123', '123'),
                     ('123', '124', '125'),
                     ('123', '123[4-5]', '123[4-6]'),
                     ('123[4-5]', '123[2-7]', '123[1-8]'),
                     ('123', '[2-3]', '4')) as t(a, b, c)
      ) as x;

-- set operations
select a, b, a | b as union, a & b as intersect
  from  (select a::prefix_range, b::prefix_range
           from (values('123', '123'),
                       ('123', '124'),
                       ('123', '123[4-5]'),
                       ('123[4-5]', '123[2-7]'),
                       ('123', '[2-3]')) as t(a, b)
        ) as x;

-- casting to and from text
select prefix_range('123');
select prefix_range('123[4-5]');
select prefix_range('[2-3]');
select prefix_range('123', '4', '5');
select length('12345'::prefix_range);
select length('12345[]'::prefix_range);
select length('123[4-5]'::prefix_range);
select length('1234'::prefix_range);
\dC *prefix*
