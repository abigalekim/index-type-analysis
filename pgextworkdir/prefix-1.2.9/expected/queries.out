-- operators
select a, b,
   a <= b as "<=", a < b as "<", a = b as "=", a <> b as "<>", a >= b as ">=", a > b as ">",
   a @> b as "@>", a <@ b as "<@", a && b as "&&"
from  (select a::prefix_range, b::prefix_range
         from (values('123', '123'),
                     ('123', '124'),
                     ('123', '123[4-5]'),
                     ('123[4-5]', '123[2-7]'),
                     ('123', '[2-3]')) as t(a, b)
      ) as x;
    a     |    b     | <= | < | = | <> | >= | > | @> | <@ | && 
----------+----------+----+---+---+----+----+---+----+----+----
 123      | 123      | t  | f | t | f  | t  | f | t  | t  | t
 123      | 124      | t  | t | f | t  | f  | f | f  | f  | f
 123      | 123[4-5] | t  | t | f | t  | f  | f | t  | f  | t
 123[4-5] | 123[2-7] | f  | f | f | t  | t  | t | f  | t  | t
 123      | [2-3]    | t  | t | f | t  | f  | f | f  | f  | f
(5 rows)

-- transitivity
select a, b, c, a <= b as "a <= b", b <= c as "b <= c", a <= c as "a <= c"
from  (select a::prefix_range, b::prefix_range, c::prefix_range
         from (values('123', '123', '123'),
                     ('123', '124', '125'),
                     ('123', '123[4-5]', '123[4-6]'),
                     ('123[4-5]', '123[2-7]', '123[1-8]'),
                     ('123', '[2-3]', '4')) as t(a, b, c)
      ) as x;
    a     |    b     |    c     | a <= b | b <= c | a <= c 
----------+----------+----------+--------+--------+--------
 123      | 123      | 123      | t      | t      | t
 123      | 124      | 125      | t      | t      | t
 123      | 123[4-5] | 123[4-6] | t      | t      | t
 123[4-5] | 123[2-7] | 123[1-8] | f      | f      | f
 123      | [2-3]    | 4        | t      | t      | t
(5 rows)

-- set operations
select a, b, a | b as union, a & b as intersect
  from  (select a::prefix_range, b::prefix_range
           from (values('123', '123'),
                       ('123', '124'),
                       ('123', '123[4-5]'),
                       ('123[4-5]', '123[2-7]'),
                       ('123', '[2-3]')) as t(a, b)
        ) as x;
    a     |    b     |  union   | intersect 
----------+----------+----------+-----------
 123      | 123      | 123      | 123
 123      | 124      | 12[3-4]  | 
 123      | 123[4-5] | 123      | 123[4-5]
 123[4-5] | 123[2-7] | 123[2-7] | 123[4-5]
 123      | [2-3]    | [1-3]    | 
(5 rows)

-- casting to and from text
select prefix_range('123');
 prefix_range 
--------------
 123
(1 row)

select prefix_range('123[4-5]');
 prefix_range 
--------------
 123[4-5]
(1 row)

select prefix_range('[2-3]');
 prefix_range 
--------------
 [2-3]
(1 row)

select prefix_range('123', '4', '5');
 prefix_range 
--------------
 123[4-5]
(1 row)

select length('12345'::prefix_range);
 length 
--------
      5
(1 row)

select length('12345[]'::prefix_range);
 length 
--------
      5
(1 row)

select length('123[4-5]'::prefix_range);
 length 
--------
      4
(1 row)

select length('1234'::prefix_range);
 length 
--------
      4
(1 row)

\dC *prefix*
                     List of casts
 Source type  | Target type  |   Function   | Implicit? 
--------------+--------------+--------------+-----------
 prefix_range | text         | text         | no
 text         | prefix_range | prefix_range | yes
(2 rows)

